cmake/common/CmakeHelperFunctions.cmake

```diff
@@ -140,6 +140,8 @@ function(get_if_link_libraries libs dirs target)
             if ("${_library}" MATCHES "LINK_ONLY")
                 string(REGEX REPLACE "\\$<LINK_ONLY:" "" __fix ${_library})
                 string(REGEX REPLACE ">$" "" _library ${__fix})
+                message(AUTHOR_WARNING "${_library} is marked LINK_ONLY and is now skipped, if you experience link issues out of the blue; start here. :-)")
+                continue()
```

Comment:
yes this changes skipping the error I have seen without adding find_package in public scope, that's great :)